source/client/options_impl.cc

```diff
@@ -371,12 +371,13 @@ CommandLineOptionsPtr OptionsImpl::toCommandLineOptions() const {
     // TODO(oschaaf): expose append option in CLI? For now we just set.
     header_value_option->mutable_append()->set_value(false);
     auto request_header = header_value_option->mutable_header();
-    std::vector<std::string> split_header = absl::StrSplit(
-        header, ':',
-        absl::SkipWhitespace()); // TODO(oschaaf): maybe throw when we find > 2 elements.
-    request_header->set_key(split_header[0]);
-    if (split_header.size() == 2) {
-      request_header->set_value(split_header[1]);
+    auto pos = header.find(':');
+    if (pos != std::string::npos) {
+      request_header->set_key(header.substr(0, pos));
```

Comment:
(optional) Seems that we are changing a behavior related to whitespace here, the previous code used the absl::SkipWhitespace() predicate. I am unsure whether that was required or if the change is intentional.

Would it be useful to add a test case with whitespace to capture the "correct" behavior that we want to keep? Please let me know if there already is one that I missed.

Reply:
@mum4k , this change will retain whitespaces, which are allowed in HTTP header values.

The `RequestHeaderValueWithColonsAndSpaces` test uses a header value that has an embedded space and expects it to be preserved.

Reply:
Thank you for the clarification.